presto-router/src/main/java/com/facebook/presto/router/cluster/ClusterManager.java

```diff
@@ -41,23 +46,44 @@
 
 public class ClusterManager
 {
-    private final Map<String, GroupSpec> groups;
-    private final List<SelectorRuleSpec> groupSelectors;
-    private final SchedulerType schedulerType;
-    private final Scheduler scheduler;
-    private final HashMap<String, HashMap<URI, Integer>> serverWeights = new HashMap<>();
+    private final RouterConfig routerConfig;
+    private Map<String, GroupSpec> groups;
+    private List<SelectorRuleSpec> groupSelectors;
+    private SchedulerType schedulerType;
+    private final ScheduledExecutorService scheduledExecutorService;
+    private Scheduler scheduler;
+    private HashMap<String, HashMap<URI, Integer>> serverWeights = new HashMap<>();
+    private final AtomicLong lastConfigUpdate = new AtomicLong();
 
     @Inject
-    public ClusterManager(RouterConfig config)
+    public ClusterManager(RouterConfig config, @ForClusterManager ScheduledExecutorService scheduledExecutorService)
     {
-        RouterSpec routerSpec = parseRouterConfig(config)
-                .orElseThrow(() -> new PrestoException(CONFIGURATION_INVALID, "Failed to load router config"));
+        this.routerConfig = config;
+        this.scheduledExecutorService = scheduledExecutorService;
+        initializeRouterConfigSpec(this.routerConfig);
+    }
+
+    @PostConstruct
+    public void startConfigReloadTask() throws Exception
+    {
+        File routerConfigFile = new File(routerConfig.getConfigFile());
+        scheduledExecutorService.scheduleAtFixedRate(() -> {
+            long newConfigUpdateTime = routerConfigFile.lastModified();
+            if (lastConfigUpdate.get() != newConfigUpdateTime) {
+                initializeRouterConfigSpec(this.routerConfig);
```

Comment:
If `routerConfig` is final, how would this actually update the configuration? This is just passing the same `routerConfig` object every time